src/charm.py

```diff
@@ -110,6 +188,47 @@ def __init__(self, *args):
         self.framework.observe(
             self.on.validate_configuration_action, self._on_validate_configuration
         )
+        self.framework.observe(self.resources_patch.on.patch_failed, self._on_k8s_patch_failed)
+
+    @classmethod
+    def _limits_to_requests(cls, resource_limits: ResourceSpecDict):
+        """Calculate "requests" from a "limits" dict.
+
+        By default, we set the "requests" portion of the resource limits to a sensible value, while
```

Comment:
Is there any thing specific to Prometheus in the logic of this function ? If not could this be a method in the Kubernetes resource patch library ?